### DISCUSSION TITLE
Implement `tabBarHidden`

### DISCUSSION THREAD

**Discussion by okwasniewski** [callstack/react-native-bottom-tabs#214] [Ideas]
Goal of this issue is to implement `tabBarHidden` config option to be able to hide the tab bar on certain routes.

**Comment by anudit** [comment]
@okwasniewski Looking forward to this

**Comment by cesargdm** [comment]
Expo implements it with `options={{ href: null }}` https://docs.expo.dev/router/advanced/tabs/#hiding-a-tab

**Comment by okwasniewski** [comment]
Hey @cesargdm, `options={{ href: null }}` is used for hiding a screen not the tab bar. 

The screen hiding is also supported using `tabBarItemHidden`, supporting `href: null` is possible but would require creating an expo specific package with the navigator (they add it on top of React Navigation tabs too).

**Comment by rochajulian** [comment]
Any update in this? The way that I used to do it on Expo is as follows:
```
<Tabs.Screen
    name="index"
    options={{
        href: null,
        tabBarStyle: {
            display: "none"
        }
    }}
/>
```      
  This would effectively hide the screen from the tab bar AND the tab bar on that screen.

**Comment by okwasniewski** [comment]
I'm moving this to discussions because all feature requests will now be under the "Ideas 💡" tab in GitHub discussions.

**Comment by guillaumehcht** [comment]
Looking forward to this as well! As I am getting issues with fixed button at the bottom of a Modal moving down after a couple of seconds after rendering and it seems it's the TabBar messing with the layout :)

I guess if if I can hide the tabBar specifically on that screen that would solve the issue? Not sure.

**Comment by anckaertv** [comment]
Would love this new feature as well !! :)

**Comment by okwasniewski** [comment]
Hey, 

After working on this feature (which was partially introduced in #273) I decided to not add this prop. 

However, in the upcoming version you will be able to still hide the tab bar by passing `tabBar={() => null}` 

Dynamically hiding/showing tab bar was causing jumps to the content triggering re-layout. That's why its also not recommended to do this with JS Tabs. 

If you want to hide the tab bar without jumps of the content, use this guide: https://reactnavigation.org/docs/hiding-tabbar-in-screens

**Comment by anudit** [comment]
@okwasniewski The gmail app on android does this pretty smoothly.


https://github.com/user-attachments/assets/e0f2df7b-1304-425f-b0b6-159d270cf9fa
